### PR TITLE
fix legend show/hide all, remove and reload layer

### PR DESCRIPTION
**Pull request by elsa-huang**
- fix show/hide all, collapse/expand all, layer remove and reload functionalities
- make `LegendHeaderV` and `LegendOptionsV` use `defineComponent`

[Demo](http://ramp4-app.azureedge.net/demo/users/elsa-huang/legend-fixture/host/index.html#)

### CODE REVIEW

File: packages/ramp-core/src/fixtures/legend/components/legend-options.vue [hunk 1]
```diff
@@ -120,84 +120,90 @@
...
+         * Toggles metadata panel to open/close for the LegendItem.
+         */
+        toggleMetadata() {
+            if (this.legendItem._controlAvailable(Controls.Metadata)) {
+                // TODO: toggle metadata panel through API/store call
+                // this.$iApi.event.emit('metadata/open', {
+                //     type: 'html',
+                //     layer: 'Sample Layer Name',
+                //     url:
+                //         'https://ryan-coulson.com/RAMPMetadataDemo.html'
```
Comment (yileifeng) [inline packages/ramp-core/src/fixtures/legend/components/legend-options.vue:175]: uncomment this for metadata panel to work (probably just a small rebase error)

File: packages/ramp-core/src/fixtures/legend/components/legend-options.vue [hunk 2]
```diff
@@ -120,84 +120,90 @@
...
-     * Display symbology stack for the layer.
-     */
-    toggleSymbology(): void {
-        if (this.legendItem._controlAvailable(Controls.Symbology)) {
-            this.legendItem.displaySymbology = !this.legendItem.displaySymbology;
-        }
-    }
+export default defineComponent({
+    name: 'LegendOptionsV',
+    props: ['legendItem'],
```
Comment (yileifeng) [inline packages/ramp-core/src/fixtures/legend/components/legend-options.vue:131]: let's keep the typing of the props if possible, i think this works: 
```
props: {
    legendItem: LegendEntry
}
```

File: packages/ramp-core/src/fixtures/legend/components/legend-options.vue (hunk 1 above)
Comment (elsa-huang) [reply packages/ramp-core/src/fixtures/legend/components/legend-options.vue]: oops, donethanks 👍

File: packages/ramp-core/src/fixtures/legend/components/legend-options.vue (hunk 2 above)
Comment (elsa-huang) [reply packages/ramp-core/src/fixtures/legend/components/legend-options.vue]: donethanks 👍

**Review by RyanCoulsonCA — APPROVED** [review]

**Review by yileifeng — APPROVED** [review]

**Review by sharvenp — APPROVED** [review]

**Pull request merged**